Src/ILGPU.Algorithms/TempViewManager.cs

```diff
@@ -86,6 +87,24 @@ public ArrayView<T> Allocate<T>(int length)
             if (NumInts + viewLength > TempView.Length)
                 throw new ArgumentOutOfRangeException(ParamName);
 
+            // Ensure correct alignment when allocating types larger than a single int.
+            // NB: Structs are assumed to be int-aligned.
+            if (typeof(T).IsPrimitive)
+            {
+                var sizeOfAllocateT = Interop.SizeOf<T>();
+                var sizeOfInt = Interop.SizeOf<int>();
+                var allocationByteOffset = NumInts * sizeOfInt;
+                if (sizeOfAllocateT > sizeOfInt &&
```

Comment:
QUESTION: What is the correct alignment rule?

RadixSort kernels appear to work just fine, but had to be changed because it did not follow the alignment rule as defined in this code.